src/components/ClusterNodes/NodeList.tsx

```diff
@@ -0,0 +1,526 @@
+import { appSelectorStyle, DropdownIndicator } from '../AppSelector/AppSelectorUtil'
+import { OptionType } from '../app/types'
+import NodeListSearchFilter from './NodeListSearchFliter'
+import { OrderBy } from '../app/list/types'
+import ClusterNodeEmptyState from './ClusterNodeEmptyStates'
+import Tippy from '@tippyjs/react'
+
+export default function NodeList() {
+    const match = useRouteMatch()
+    const history = useHistory()
+    const [loader, setLoader] = useState(false)
+    const [searchText, setSearchText] = useState('')
+    const [clusterCapacityData, setClusterCapacityData] = useState<ClusterCapacityType>(null)
+    const [lastDataSyncTimeString, setLastDataSyncTimeString] = useState('')
+    const [lastDataSync, setLastDataSync] = useState(false)
+    const [collapsedErrorSection, setCollapsedErrorSection] = useState<boolean>(true)
+    const { clusterId } = useParams<{ clusterId: string }>()
+    const [clusterList, setClusterList] = useState<OptionType[]>([])
+    const [selectedCluster, setSelectedCluster] = useState<OptionType>({
+        label: '',
+        value: '',
+    })
+    const defaultVersion = { label: 'K8s version: Any', value: 'K8s version: Any' }
+    const [clusterErrorTitle, setClusterErrorTitle] = useState('')
+    const [clusterErrorList, setClusterErrorList] = useState<string[]>([])
+    const [flattenNodeList, setFlattenNodeList] = useState<object[]>([])
+    const [filteredFlattenNodeList, setFilteredFlattenNodeList] = useState<object[]>([])
+    const [searchedLabelMap, setSearchedLabelMap] = useState<Map<string, string>>(new Map())
+    const [selectedVersion, setSelectedVersion] = useState<OptionType>(defaultVersion)
+    const [selectedSearchTextType, setSelectedSearchTextType] = useState<string>('')
+    const [sortByColumn, setSortByColumn] = useState<columnMetadataType>(columnMetadata[0])
+    const [sortOrder, setSortOrder] = useState<string>(OrderBy.ASC)
+    const [noResults, setNoResults] = useState(false)
+    const [appliedColumns, setAppliedColumns] = useState<MultiValue<columnMetadataType>>([])
+
+    useEffect(() => {
+        let appliedColumnsFromLocalStorage
+        const _defaultColumns = columnMetadata.filter((columnData) => columnData.isDefault)
+        if (typeof Storage !== 'undefined') {
+            if (!localStorage.appliedColumns) {
+                localStorage.appliedColumns = JSON.stringify(_defaultColumns)
+            } else {
+                try {
+                    appliedColumnsFromLocalStorage = JSON.parse(localStorage.appliedColumns)
+                } catch (error) {}
+            }
+        }
+        setAppliedColumns(appliedColumnsFromLocalStorage || _defaultColumns)
+    }, [])
+
+    const flattenObject = (ob: Object): Object => {
+        var toReturn = {}
+
+        for (var i in ob) {
+            if (!ob.hasOwnProperty(i)) continue
+
+            if (typeof ob[i] == 'object' && ob[i] !== null && !Array.isArray(ob[i])) {
```

Comment:
use `===`

Reply:
Also, take the value of ob[i] in some var & then perform all ops.

Reply:
Done